### ISSUE TITLE
我怎样知道我的num_gpu_cores

### ISSUE THREAD

**Issue by lu161513** [HaoyuHu/bert-multi-gpu#14]
这个是显卡的哪个参数呀  num_gpu_cores

**Comment by haoyuhu** [comment]
> 这个是显卡的哪个参数呀 num_gpu_cores

`num_gpu_cores` is the number of GPU.

You can get ALL GPUs:
```python
from tensorflow.python.client import device_lib as _device_lib
local_device_protos = _device_lib.list_local_devices()
gpus = [x.name for x in local_device_protos if x.device_type == 'GPU']
print(gpus)

# results:
# ['/device:GPU:0', '/device:GPU:1']
```

Or

```shell
nvidia-smi

# results:
# +-----------------------------------------------------------------------------+
# | NVIDIA-SMI 430.14       Driver Version: 430.14       CUDA Version: 10.2     |
# |-------------------------------+----------------------+----------------------+
# | GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
# | Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
# |===============================+======================+======================|
# |   0  GeForce GTX 108...  Off  | 00000000:03:00.0 Off |                  N/A |
# | 22%   40C    P8    18W / 250W |    145MiB / 11176MiB |      0%      Default |
# +-------------------------------+----------------------+----------------------+
# |   1  GeForce GTX 108...  Off  | 00000000:04:00.0 Off |                  N/A |
# | 21%   35C    P8    10W / 250W |    145MiB / 11178MiB |      0%      Default |
# +-------------------------------+----------------------+----------------------+
#                                                                                
# +-----------------------------------------------------------------------------+
# | Processes:                                                       GPU Memory |
# |  GPU       PID   Type   Process name                             Usage      |
# |=============================================================================|
# |    0     47017      C   .../local/environment/anaconda3/bin/python   135MiB |
# |    1     47017      C   .../local/environment/anaconda3/bin/python   135MiB |
# +-----------------------------------------------------------------------------+
```

**Comment by lu161513** [comment]
还有个疑问，
I1112 20:14:40.599461 140252801017664 run_custom_classifier.py:1102] ***** Eval results *****
I1112 20:14:40.599556 140252801017664 run_custom_classifier.py:1104]   eval_accuracy = 0.9807692
I1112 20:14:40.601030 140252801017664 run_custom_classifier.py:1104]   eval_loss = 0.13390507
I1112 20:14:40.601123 140252801017664 run_custom_classifier.py:1104]   global_step = 10658
I1112 20:14:40.601165 140252801017664 run_custom_classifier.py:1104]   loss = 0.12894636
I1112 20:14:40.605823 140252801017664 run_custom_classifier.py:595] Writing example 0 of 211
I1112 20:14:40.606251 140252801017664 run_custom_classifier.py:569] *** Example ***


怎样可以添加f1,precision,recall?
我跑官方的样例是有的：
I1023 23:32:39.085123 139676537534272 run_classifier_serving.py:1024] ***** Eval results *****
I1023 23:32:39.085174 139676537534272 run_classifier_serving.py:1026]   eval_accuracy = 0.86040765
I1023 23:32:39.085299 139676537534272 run_classifier_serving.py:1026]   eval_f1 = 0.95126426
I1023 23:32:39.085345 139676537534272 run_classifier_serving.py:1026]   eval_loss = 1.3762913
I1023 23:32:39.085381 139676537534272 run_classifier_serving.py:1026]   eval_precision = 0.95021963
I1023 23:32:39.085416 139676537534272 run_classifier_serving.py:1026]   eval_recall = 0.9523111
I1023 23:32:39.085456 139676537534272 run_classifier_serving.py:1026]   global_step = 7590
I1023 23:32:39.085492 139676537534272 run_classifier_serving.py:1026]   loss = 1.372054

我看代码中也只有accuracy = tf.metrics.accuracy 这个呀，其他哪儿来的

**Comment by haoyuhu** [comment]
> 还有个疑问，
> I1112 20:14:40.599461 140252801017664 run_custom_classifier.py:1102] ***** Eval results *****
> I1112 20:14:40.599556 140252801017664 run_custom_classifier.py:1104] eval_accuracy = 0.9807692
> I1112 20:14:40.601030 140252801017664 run_custom_classifier.py:1104] eval_loss = 0.13390507
> I1112 20:14:40.601123 140252801017664 run_custom_classifier.py:1104] global_step = 10658
> I1112 20:14:40.601165 140252801017664 run_custom_classifier.py:1104] loss = 0.12894636
> I1112 20:14:40.605823 140252801017664 run_custom_classifier.py:595] Writing example 0 of 211
> I1112 20:14:40.606251 140252801017664 run_custom_classifier.py:569] *** Example ***
> 
> 怎样可以添加f1,precision,recall?
> 我跑官方的样例是有的：
> I1023 23:32:39.085123 139676537534272 run_classifier_serving.py:1024] ***** Eval results *****
> I1023 23:32:39.085174 139676537534272 run_classifier_serving.py:1026] eval_accuracy = 0.86040765
> I1023 23:32:39.085299 139676537534272 run_classifier_serving.py:1026] eval_f1 = 0.95126426
> I1023 23:32:39.085345 139676537534272 run_classifier_serving.py:1026] eval_loss = 1.3762913
> I1023 23:32:39.085381 139676537534272 run_classifier_serving.py:1026] eval_precision = 0.95021963
> I1023 23:32:39.085416 139676537534272 run_classifier_serving.py:1026] eval_recall = 0.9523111
> I1023 23:32:39.085456 139676537534272 run_classifier_serving.py:1026] global_step = 7590
> I1023 23:32:39.085492 139676537534272 run_classifier_serving.py:1026] loss = 1.372054
> 
> 我看代码中也只有accuracy = tf.metrics.accuracy 这个呀，其他哪儿来的

If you need these metrics, you can add them yourself.
REF: 
https://github.com/Vincent131499/TextClassifier_BERT/blob/master/run_classifier_serving.py#L760
https://github.com/HaoyuHu/bert-multi-gpu/blob/master/run_custom_classifier.py#L748

**Comment by haoyuhu** [comment]
The above metrics will be added in the next version, related PR: https://github.com/HaoyuHu/bert-multi-gpu/pull/17.
